ChatSecure/Classes/Controllers/XMPP/OTRXMPPManager.m

```diff
@@ -1383,6 +1383,56 @@ - (void) addBuddy:(OTRXMPPBuddy *)newBuddy
     [self addBuddies:@[newBuddy]];
 }
 
+- (OTRXMPPBuddy *)addBuddy:(XMPPJID *)jid displayName:(NSString *)displayName {
+    __block OTRXMPPBuddy *buddy = nil;
+    [self.databaseConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+        buddy = [self addBuddy:jid displayName:displayName transaction:transaction];
+    }];
+    return buddy;
+}
+
+- (OTRXMPPBuddy *)addBuddy:(XMPPJID *)jid displayName:(NSString *)displayName transaction:(YapDatabaseReadWriteTransaction *)transaction{
+    
+    OTRXMPPBuddy *buddy = [OTRXMPPBuddy fetchBuddyWithJid:jid accountUniqueId:self.account.uniqueId transaction:transaction];
+    if (!buddy) {
+        buddy = [[OTRXMPPBuddy alloc] init];
+        buddy.username = [jid bare];
+        buddy.accountUniqueId = self.account.uniqueId;
+    } else {
+        buddy = [buddy copy];
+    }
+    
+    // Update display name
+    if (displayName && [displayName length] > 0) {
+        buddy.displayName = displayName;
+    }
+    
+    buddy.trustLevel = BuddyTrustLevelRoster;
+    if ([buddy askingForApproval]) {
+        // We have an incoming subscription request, just answer that!
+        // TODO - use queue for this as well!
+        [buddy setAskingForApproval:NO];
+        [buddy saveWithTransaction:transaction];
+        
+        [self.xmppRoster acceptPresenceSubscriptionRequestFrom:jid andAddToRoster:YES];
+        
+        // Cleanup of old code. We no longer use the OTRXMPPPresenceSubscriptionRequest class,
+        // so if we have an old entry for this JID, just delete it
+        OTRXMPPPresenceSubscriptionRequest *presenceRequest = [OTRXMPPPresenceSubscriptionRequest fetchPresenceSubscriptionRequestWithJID:[jid bare] accontUniqueId:self.account.uniqueId transaction:transaction];
```

Comment:
You could also just do a `[transaction removeAllObjectsFromCollection: OTRXMPPPresenceSubscriptionRequest.collection]` during the database setup.

Reply:
Yeah, seems cleaner.